fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/codec/LuceneOptimizedPointsFormat.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * LuceneOptimizedPointsFormat.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2023 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.lucene.codec;
+
+import org.apache.lucene.codecs.PointsFormat;
+import org.apache.lucene.codecs.PointsReader;
+import org.apache.lucene.codecs.PointsWriter;
+import org.apache.lucene.codecs.lucene86.Lucene86PointsFormat;
+import org.apache.lucene.codecs.lucene86.Lucene86PointsReader;
+import org.apache.lucene.index.PointValues;
+import org.apache.lucene.index.SegmentReadState;
+import org.apache.lucene.index.SegmentWriteState;
+
+import java.io.IOException;
+
+/**
+ * Lazy Reads the PointsFormat to limit the amount of bytes returned
+ * from FDB.
+ *
+ */
+public class LuceneOptimizedPointsFormat extends PointsFormat {
+
+    PointsFormat pointsFormat;
+
+    public LuceneOptimizedPointsFormat() {
+        this.pointsFormat = new Lucene86PointsFormat();
+    }
+
+    @Override
+    public PointsWriter fieldsWriter(final SegmentWriteState state) throws IOException {
+        return pointsFormat.fieldsWriter(state);
+    }
+
+    @Override
+    public PointsReader fieldsReader(final SegmentReadState state) throws IOException {
+        return new LazyPointsReader(state);
+    }
+
+    private static class LazyPointsReader extends PointsReader {
+        private final SegmentReadState state;
+        private PointsReader pointsReader;
+
+        private LazyPointsReader(final SegmentReadState state) {
+            this.state = state;
+
+        }
+
+        private synchronized void checkInit() throws IOException {
+            if (pointsReader == null) {
+                pointsReader = new Lucene86PointsReader(state);
```

Comment:
This may be replaced by a call to the `pointsFormat.fieldsReader(state)`, right?